### PR TITLE
Add ecosystem page

### LINKED ISSUE #4704: Add a page to the website documenting Contour's integrations with other tools
It'd be great to have a page on the website documenting/linking to docs on Contour's integrations with various other tools. Some I know of off the top of my head:
- [Open Service Mesh](https://release-v1-2.docs.openservicemesh.io/docs/demos/ingress_contour/)
- [Linkerd](https://linkerd.io/2.12/tasks/using-ingress/#contour)
- [knative serving](https://knative.dev/docs/install/yaml-install/serving/install-serving-with-yaml/#install-a-networking-layer)
- [ArgoCD](https://argo-cd.readthedocs.io/en/stable/operator-manual/ingress/#contour)
- ...

ref. #2204
ref. #2920
ref. #2089
ref. #2506

### LINKED ISSUE #5280: document the argo-rollouts plugin for httpproxy
Hey @skriss @sunjayBhatia our team just release [a plugin](https://github.com/argoproj-labs/rollouts-plugin-trafficrouter-contour) for  `contour/httpproxy` to let it can work with `argo-rollouts`, so whether we need to make a note at here?  If the answer is yes, where is the right place to do that?  Thanks.

### PR DESCRIPTION
**Pull request by izturn**
Signed-off-by: Gang Liu [gang.liu@daocloud.io](mailto:gang.liu@daocloud.io)

fix #5280 
fix #4704

### CODE REVIEW

**Comment by izturn** [comment]
ping

**Comment by izturn** [comment]
ping

**Comment by tsaarni** [comment]
Just a question: was the conclusion in #5280 and #4704 to add an ecosystem web page on projectcontour.io?

**Comment by tsaarni** [comment]
If aiming for a web page on the site, shouldn't we create the file under `site/content/` directory structure? It is the source for the website, which is generated with hugo. It can be run locally as well 
https://github.com/projectcontour/contour/blob/6a1ee75be5e1c7bbb11c1790ed55123946d1587e/Makefile#L291-L292

**Comment by izturn** [comment]
TBH, I am not sure where is the right place, put it here because of this @sunjayBhatia's comment: https://github.com/projectcontour/contour/issues/5280#issuecomment-1511502887, so I think the `ECOSYSTEM.md` should be the sibling of `ADOPTERS.md`, @sunjayBhatia any advice?

**Comment by izturn** [comment]
ping @sunjayBhatia

**Comment by skriss** [comment]
Personally I would prefer having this page be on the website.

**Comment by sunjayBhatia** [comment]
Yeah I think we maybe want to move/add the adopters file to the website, along with the ecosystem/integrations so they're both a bit more discoverable by search engines etc.

**Comment by izturn** [comment]
@tsaarni @skriss @sunjayBhatia  PTAL

**Review by sunjayBhatia — CHANGES_REQUESTED** [review]
deploy preview pages:
- https://deploy-preview-5537--projectcontour.netlify.app/adopters/
- https://deploy-preview-5537--projectcontour.netlify.app/ecosystem/

With the current formatting, looks like the pages are being rendered as blog posts with an invalid date, we'll have to add something like this to the top of the page: https://github.com/projectcontour/contour/blob/9ad7ac550a3a5861d9200a0f427023c3e9fe6c90/site/content/resources/compatibility-matrix.md#L1-L4

also looks like the relative links to images in the adopters page no longer work (like https://github.com/projectcontour/contour/blob/9ad7ac550a3a5861d9200a0f427023c3e9fe6c90/ADOPTERS.md#L6)

**Comment by izturn** [comment]
> It would be great to link these pages to each other and link them on the main index or at least in the table of contents of the site

@sunjayBhatia, pls tell me how to do it, thx

**Comment by izturn** [comment]
@sunjayBhatia  ping

**Comment by skriss** [comment]
I'd suggest adding them as "resources" under `site/content/resources`, and adding them to the table of contents in https://github.com/projectcontour/contour/blob/6cbb66c96bb501955693932c181d42d1f37b4ec0/site/data/docs/main-toc.yml#L118-L131

File: site/content/ECOSYSTEM.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+layout: page
+---
+
+This is a list of Contour related resources. If you have a suggestion on what to add, please don't hesitate to [submit a pull request][2]!
+
+## Help
+
+### Creating issue
+
+-   [Contour/issues][1]
```
Comment (skriss) [inline site/content/ECOSYSTEM.md:12]: IMO we can drop this, this is off-topic for this page.

**Comment by sunjayBhatia** [comment]
Updated deploy preview pages:
- https://deploy-preview-5537--projectcontour.netlify.app/resources/ecosystem/
- https://deploy-preview-5537--projectcontour.netlify.app/resources/adopters/

File: site/content/resources/adopters.md [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
+---
+title: Contour Adopters
+layout: page
+---
 
 If you're using Contour and want to add your organization to this list, please
 [submit a pull request][1]!
 
-<a href="https://knative.dev" border="0" target="_blank"><img alt="knative.dev" src="site/img/adopters/knative.svg" height="50"></a>
+<a href="https://knative.dev" target="_blank"><img alt="knative.dev" src="../../img/adopters/knative.svg" height="50"></a>
```
Comment (sunjayBhatia) [inline site/content/resources/adopters.md:9]: hm, looks like we have these files in two places now, would be good to consolidate the images into one location

see:
- https://github.com/izturn/contour/tree/doc/ecosystem/site/img/adopters
- https://github.com/izturn/contour/tree/doc/ecosystem/site/themes/contour/static/img/adopters

Comment (izturn) [reply site/content/resources/adopters.md:9]: i add it `../themes`, because the first place will not work, so may i delete the first one?

**Review by skriss — APPROVED** [review]
thanks @izturn, LGTM

**Review by sunjayBhatia — APPROVED** [review]

**Pull request merged**